auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -310,16 +312,51 @@ public TypeSpec createTypeAdapter(ClassName className, ClassName autoValueClassN
     return classBuilder.build();
   }
 
+  private static void buildParameterizedToken(MethodSpec.Builder constructor, TypeName typeArg,
+                                              List<TypeVariableName> typeParams, boolean first) {
+    if (!first) {
+      constructor.addCode(", ");
+    }
+    if (typeArg instanceof ParameterizedTypeName) {
+      ParameterizedTypeName paramTypeArg = (ParameterizedTypeName) typeArg;
+      constructor.addCode("$T.newParameterizedTypeWithOwner(null, $T.class", $Gson$Types.class, paramTypeArg.rawType);
```

Comment:
Copy it out for yourself, that's what I did https://github.com/evant/gsonvalue/blob/master/gsonvalue/src/main/java/me/tatarka/gsonvalue/internal/Types.java